WalletWasabi/Blockchain/Keys/HdPubKey.cs

```diff
@@ -2,15 +2,18 @@
 using Newtonsoft.Json;
 using System;
 using System.Linq;
+using WalletWasabi.Bases;
 using WalletWasabi.Blockchain.Analysis.Clustering;
 using WalletWasabi.Helpers;
 using WalletWasabi.JsonConverters;
 
 namespace WalletWasabi.Blockchain.Keys
 {
 	[JsonObject(MemberSerialization.OptIn)]
-	public class HdPubKey : IEquatable<HdPubKey>
+	public class HdPubKey : NotifyPropertyChangedBase, IEquatable<HdPubKey>
```

Comment:
If you do this do this properly and make it for all the possible properties those can change.